CrowsZA/functions/fn_fireSupport.sqf

```diff
@@ -0,0 +1,205 @@
+    diag_log _msg;
+    breakOut "Main";
+};
+
+// open dialog
+//ZEN
+private _onConfirm =
+{
+	params ["_dialogResult","_in"];
+	_dialogResult params
+	[
+		"_type",
+		"_customType",
+		"_radius",
+		"_seconds",
+		"_delay",
+		"_salvos",
+		"_guns"
+	];
+
+	//Get in params again
+	_in params [["_pos",[0,0,0],[[]],3], ["_logic",objNull,[objNull]]];
+
+	//Check if placed on an object
+	_create = isNull _logic;
+
+	//Check if object is firesupport module
+	if(!_create)
+	then
+	{
+		if(!(_logic isKindOf "Logic"))
+		then
+		{
+			_create = true;
+		}
+		else {
+			_mod = _logic getVariable ["_module", ""];
+			if(_mod != "firesupport")
+			then
+			{
+				_create = true;
+			};
+		}
+	};
+
+	//Create new module if not placed on existing one
+	if(_create)
+	then{
+		_logicCenter = createCenter sideLogic;
+		_logicGroup = createGroup _logicCenter;
+		_logic = _logicGroup createUnit ["Logic", _pos, [], 0, "NONE"];
+		_logic setVariable ["_module", "firesupport"];
+		["zen_common_addObjects", [[_logic], objNull]] call CBA_fnc_serverEvent;
+	};
+
+	//Use specific cfgAmmo if filled
+	if(_customType != "")
+	then{
+		_type = _customType;
+	};
+	diag_log "guns:";
+	diag_log str _guns;
+	//Set variables to logic gameobject
+	_logic setVariable ["_type", _type];
+	_logic setVariable ["_radius", _radius];
+	_logic setVariable ["_seconds", _seconds];
+	_logic setVariable ["_salvos", round _salvos];
+	_logic setVariable ["_guns", round _guns];
+	
+
+	_spawnBarrage = 
+	{
+		_logic = (_this select 0);
+		_code = (_this select 1);
+
+		if(!(isNull _logic))
+		then{
+			//Read values from logic gameobject
+			_pos = getPos _logic;
+			_type = _logic getVariable "_type";
+			_radius = _logic getVariable "_radius";
+			_seconds = _logic getVariable "_seconds";
+			_salvos = _logic getVariable "_salvos";
+			_guns = _logic getVariable "_guns";
+
+			//Spawn salvo with values
+			[_pos, _type, _radius, _guns, [0, 0.5]] spawn BIS_fnc_fireSupportVirtual;
+
+			//Wait till next salvo
+			sleep _seconds;
+			
+			//Check if more salvos should be spawned
+			if (_salvos > 1) 
+			then { 
+				//Reduces amount of remaining salvos and spawn next
+				_logic setVariable ["_salvos", _salvos -1];
+				[_logic, _code] spawn _code; 
+			} 
+			else {if (_salvos == 1) 
+				then{
+					//Delete if all salvos are done
+					deleteVehicle _logic;
+				}
+				else {if(_salvos == 0)
+					then {
+						//Don't stop if savlos is equal to 0
+						[_logic, _code] spawn _code;
+					}
+					else{
+						deleteVehicle _logic;
+					}
+				}
+			}
+		}
+	};
+
+	_delayedBarrage =
+	{
+		_logic = (_this select 0);
+		_delay = (_this select 1);
+		_code = (_this select 2);
+		
+		//Spawn barrage after delay
+		sleep _delay;
+		[_logic, _code] spawn _code;
+	};
+
+	// _spawnAreaMarker =
+	// {
+	// 	_logic = (_this select 0);
+	// 	_radius = (_this select 1);
+	// 	_pos = (_this select 2);
+		
+	// 	_obj = createSimpleObject ["\a3\Modules_F_Curator\Ordnance\surfaceMortar.p3d", _pos];
```

Comment:
``createSimpleObject`` has a ``local`` parameter. 
![image](https://user-images.githubusercontent.com/7889925/123560030-29be2b00-d7a0-11eb-80e2-5b70b7ed72d2.png)

So you should be able to do something like
```sqf
{
    // do we want it on jip if new zeus reconnect? probably. Issue with local creation is that we would require the cleanup to also happen for each zeus. 
    // So the part that deletes the object, would have to likewise do remoteExec for each zeus.
    ["\a3\Modules_F_Curator\Ordnance\surfaceMortar.p3d", _pos] remoteExec ["createSimpleObject", _x, true]; 
} forEach allCurators;
```

Reply:
I'll give it a try and see if it works, but there is also a performance problem with the scaling itself. During my testing a radius of 500m would be an approximate scale of 12x and takes 2-3 seconds to spawn. The bigger you go the worse it gets. A different solution all together might be better.

Reply:
hmm alright. You can check and time how long a function takes using the debugconsole. Has a timings function where it repeats the code and give you the timings of it. Can try to do that with bigger scales of it. I think it will take less time as local as it only spawns it on your local pc, so only the zeus machine works, but would require test. The scaling might just not be created for performance ;-)